### ISSUE TITLE
[BUG] Confusing behaviour of `ControlledQubitUnitary`

### ISSUE THREAD

**Issue by ankit27kh** [PennyLaneAI/pennylane#2606]
### Expected behavior

When using `ControlledQubitUnitary` to make a controlled version of pre-defined gates, using `qml.matrix(qml.some_gate(param, wire))` to get the unitary of the pre-defined gate should work correctly.

### Actual behavior

The resulting gate is different for each value of the `wire` in the `qml.matrix` method.
For example:
```
@qml.qnode(dev)
def c2(wire, angle):
    qml.ControlledQubitUnitary(
        qml.matrix(qml.RY(angle, wire)),
        control_wires=[1, 2],
        wires=0,
        control_values='01'
    )
    return qml.state()


for w in [0, 1, 2]:
    print(c2(w, .5))
```
Here, the state is different for each value of wire. When it should be independent of it.
The output is:
```
[0.96891242+0.j 0.        +0.j 0.        +0.j 0.        +0.j
 0.24740396+0.j 0.        +0.j 0.        +0.j 0.        +0.j]
[0.96891242+0.j 0.        +0.j 0.24740396+0.j 0.        +0.j
 0.        +0.j 0.        +0.j 0.        +0.j 0.        +0.j]
[0.96891242+0.j 0.23971277+0.j 0.        +0.j 0.        +0.j
 0.        +0.j 0.06120872+0.j 0.        +0.j 0.        +0.j]
 ```
Different outputs which should have been same.

### Additional information

Looking at the circuits of the 3 different cases, it is clear what the issue is. When using `qml.matrix`, the gate inside this method is also being applied.
The three different circuits are:
```
0: ──RY(0.50)─╭U(M0)─┤  State
1: ───────────├C─────┤  State
2: ───────────╰C─────┤  State

0: ───────────╭U(M0)─┤  State
1: ──RY(0.50)─├C─────┤  State
2: ───────────╰C─────┤  State

0: ───────────╭U(M0)─┤  State
1: ───────────├C─────┤  State
2: ──RY(0.50)─╰C─────┤  State
```
When using an array directly as the unitary to be controlled, there is no extra gate:
```
0: ─╭U(M0)─┤  State
1: ─├C─────┤  State
2: ─╰C─────┤  State
```

Looks like the issue is with `qml.matrix`. This should have a condition that the specified gate does not apply to the Qnode.

### System information

```shell
Name: PennyLane
Version: 0.23.1
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /lib/python3.9/site-packages
Requires: autograd, autoray, semantic-version, pennylane-lightning, appdirs, scipy, retworkx, networkx, cachetools, numpy, toml
Required-by: PennyLane-qiskit, PennyLane-Lightning, PennyLane-Lightning-GPU
Platform info:           Linux-5.13.0-41-generic-x86_64-with-glibc2.31
Python version:          3.9.12
Numpy version:           1.22.3
Scipy version:           1.8.0
Installed devices:
- default.gaussian (PennyLane-0.23.1)
- default.mixed (PennyLane-0.23.1)
- default.qubit (PennyLane-0.23.1)
- default.qubit.autograd (PennyLane-0.23.1)
- default.qubit.jax (PennyLane-0.23.1)
- default.qubit.tf (PennyLane-0.23.1)
- default.qubit.torch (PennyLane-0.23.1)
- qiskit.aer (PennyLane-qiskit-0.23.0)
- qiskit.basicaer (PennyLane-qiskit-0.23.0)
- qiskit.ibmq (PennyLane-qiskit-0.23.0)
- qiskit.ibmq.circuit_runner (PennyLane-qiskit-0.23.0)
- qiskit.ibmq.sampler (PennyLane-qiskit-0.23.0)
- lightning.qubit (PennyLane-Lightning-0.23.0)
- lightning.gpu (PennyLane-Lightning-GPU-0.23.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by josh146** [comment]
Hey @ankit27kh! While I agree this is confusing, it is actually expected behaviour and unrelated to `qml.matrix`.

What is happening is that the QNode will always apply _any_ gate that is instantiated inside of it. Since `qml.RY` is instantiated inside of `qml.matrix` --- and Python always evaluates functions from 'the inside out' --- this is causing the `qml.RY` gate to be added to the circuit.

To avoid this, you can apply `qml.matrix` as a _transform_ to the gate, by using the syntax `qml.matrix(RY)(angle, wire)`:

```python
import pennylane as qml
from pennylane import numpy as np

dev = qml.device("default.qubit", wires=3)

@qml.qnode(dev)
def c2(wire, angle):
    qml.ControlledQubitUnitary(
        qml.matrix(qml.RY)(angle, wire),
        control_wires=[1, 2],
        wires=0,
        control_values='01'
    )
    return qml.state()


for w in [0, 1, 2]:
    print(qml.draw(c2)(w, .5))
```

The output will now give:

```
0: ─╭U(M0)─┤  State
1: ─├C─────┤  State
2: ─╰C─────┤  State

0: ─╭U(M0)─┤  State
1: ─├C─────┤  State
2: ─╰C─────┤  State

0: ─╭U(M0)─┤  State
1: ─├C─────┤  State
2: ─╰C─────┤  State
```

and you should find that all states have the same value.

**Comment by ankit27kh** [comment]
Thanks for the clarification @josh146

**Comment by josh146** [comment]
No worries @ankit27kh! Feel free to reopen/create a new issue if you encounter any other issues.
